### DISCUSSION TITLE
Featuring headroom in Bind The Gap

### DISCUSSION THREAD

**Discussion by chshersh** [vaclavsvejcar/headroom#63] [Show and tell]
Hi @vaclavsvejcar!

We really like `headroom` in @kowainik and we are looking forward to using it this year to update all headers ©️  
In the recently released festive edition of @bindthegap, we dedicated the section "The Wonderful Wizard of OS" to  `headroom`, spreading a good word about this wonderful tool. Just letting you know that your work is appreciated 🤗 

We hope you enjoy reading it!

* [Bind The Gap: Issue #02, December 2020](https://bindthegap.news/issues/02dec2020.html)

**Comment by xwinus** [comment]
@chshersh Wow, this is just awesome, thanks a lot! 😊 What makes me happiest about _headroom_ is that it helps someone else as well. Recently I haven't had as much time as I'd like to for _headroom_ development, but I'm still looking for cool new features to incorporate. Also I'm preparing new major version [0.4.0.0](https://github.com/vaclavsvejcar/headroom/milestone/10), I'd love to hear any feedback if you'd have some time to have a look 🙂

**Reply by vrom911** [reply]
Hey @vaclavsvejcar !
I have just battle-tested 0.4.0.0 on Summoner: https://github.com/kowainik/summoner/pull/528
The release is great!

First I had a few troubles due to regular expressions, it lead to running in the `add` mode instead of `replace`. 
But I have used the great documentation to change block-comment settings (just copied the one in there 😄 ) and it worked! 
Maybe this information needs to be highlighted more, as if you don't change your `starts-with`/`ends-with` to regex, it will silently run successfully, but not as expected. Though this information is in the migration guide, so it should be enough! 👍🏼 

But it was really nice to have all copyrights updated correctly just with one command! 🌟

**Reply by xwinus** [reply]
Hi @vrom911,
thanks a lot for you feedback. Regarding the regular expressions, I was thinking a lot whether to do this breaking change or not, but I found by using Headroom on some of my other projects that it becomes impossible to define more complex rules using just plain text matching. I'll try to highlight this information more anyway. So far when you upgrade from <0.4.0.0, it will basically force you to check the migration guide and I'll take care to make this obvious.

But still, please keep in mind that the `v0.4.0.0` is work in progress and while I'm trying my best to keep the code in `master` as much tested as possible and don't plan any more breaking changes from end user perspective for this release, I'm still doing a lot of internal changes that might eventually break something.
